### PR TITLE
Article about Apache Kafka

### PR DESCRIPTION
**Pull request by prokopsimek**

### CODE REVIEW

File: practices/kafka.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+[Confluent: Apache Kafka vs. Enterprise Service Bus (ESB) – Friends, Enemies or Frenemies?](https://www.confluent.io/blog/apache-kafka-vs-enterprise-service-bus-esb-friends-enemies-or-frenemies/)
+
+**Microservices architecture with Kafka**
+
+![Kafka Microservices](/files/kafka_microservices_correct.png)
+[Confluent: Apache Kafka vs. Enterprise Service Bus (ESB) – Friends, Enemies or Frenemies?](https://www.confluent.io/blog/apache-kafka-vs-enterprise-service-bus-esb-friends-enemies-or-frenemies/)
+
+## How to Implement Apache Kafka
+
+It's necessary to have deployed an Apache Kafka cluster including Zookeeper cluest to manage Kafka nodes. There are several libraries for programming languages to connect Kafka easilly.
```
Comment (vlasy) [inline practices/kafka.md]: Probably typo: cluest

File: practices/kafka.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+- [JMS Client](https://docs.confluent.io/current/clients/kafka-jms-client/index.html)
+
+## Common Pitfalls of the Apache Kafka
+
+- Keeping too much data
+- Old Data in Topics Not Being Deleted
+- Not balancing topics
+- Not accounting for long-term storage
+- No disaster recovery
+- No API envforcement
```
Comment (vlasy) [inline practices/kafka.md]:
```suggestion
- No API enforcement
```

File: practices/kafka.md [hunk 3]
```diff
@@ -12,27 +12,34 @@ tags:
...
+  - streaming
 covered_by_dx_scanner: false
 ---
 
 # Apache Kafka
 
 **TL;DR**
 
-aaa
+Apache Kafka is an event streaming platform initially used as messaging queue created as open-source by LinkedIn in 2011.
```
Comment (ryzzaki) [inline practices/kafka.md]: I would suggest restructuring this sentence a bit. Using `as` twice in one sentence does not sound very native to me. My rough suggestion:

Apache Kafka is an open-source event streaming platform created by LinkedIn in 2011 which initially served as a messaging queue.

Comment (prokopsimek) [reply practices/kafka.md]: Thanks!

File: practices/kafka.md (hunk 2 above)
Comment (prokopsimek) [reply practices/kafka.md]: Thanks!

File: practices/kafka.md (hunk 1 above)
Comment (prokopsimek) [reply practices/kafka.md]: Thanks! :)

File: practices/kafka.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+**TL;DR**
+
+Apache Kafka is an open-source event streaming platform created by LinkedIn in 2011 which initially served as a messaging queue.
+
+## What Is a Apache Kafka
+
+Apache Kafka is used as a high-available messaging queue. It receives messages from other services in the environment and provides it to the others.
+
+Kafka is commonly deployed as a **cluster** with 3 or more **brokers** (nodes) to have data **replicas** (backups) on other brokers.
+Kafka receives **messages** from **producers** and provides them to **consumers**. Each message is saved to a **topic** that has a name. The message can be a text, number or an object. It depends on the implementation. The topic is a category name for messages. Producers write messages to topics and consumers read messages from topics. Kafka retains all messages for a specific time and consumers are responsible to track location of these messages. Kafka topics are divided into a number of **partitions**, which contains messages in an unchangeable sequence. Each message in a partition has a specific **offset**.
```
Comment (adelkahomolova) [inline practices/kafka.md]:
```suggestion
Kafka receives **messages** from **producers** and provides them to **consumers**. Each message is saved to a **topic** that has a name. The message can be a text, number or an object. It depends on the implementation. The topic is a category name for messages. Producers write messages to topics and consumers read messages from topics. Kafka retains all messages for a specific time and consumers are responsible to track location of these messages. Kafka topics are divided into a number of **partitions**, which contain messages in an unchangeable sequence. Each message in a partition has a specific **offset**.
```

About partitions - you wrote, what it does but not what it is.

**Comment by prokopsimek** [comment]
@DStatPersonal Could you do a review, please?
